src/deno_dir.rs

```diff
@@ -182,33 +177,24 @@ impl DenoDir {
     module_name: &str,
     filename: &str,
   ) -> DenoResult<Option<CodeFetchOutput>> {
-    let extensions = ["", ".ts", ".js"];
-    for ext in extensions.iter() {
-      let filename = [filename, ext].concat();
-      let module_name = [module_name, ext].concat();
-      let p = Path::new(&filename);
-      if !p.exists() {
-        continue;
-      }
-      let media_type_filename = [&filename, ".mime"].concat();
-      let mt = Path::new(&media_type_filename);
-      let source_code = fs::read(&p)?;
-      // .mime file might not exists
-      // this is okay for local source: maybe_content_type_str will be None
-      let maybe_content_type_string = fs::read_to_string(&mt).ok();
-      // Option<String> -> Option<&str>
-      let maybe_content_type_str =
-        maybe_content_type_string.as_ref().map(String::as_str);
-      return Ok(Some(CodeFetchOutput {
-        module_name,
-        filename: filename.clone(), // TODO: no clone after NLL rfc
-        media_type: map_content_type(&p, maybe_content_type_str),
-        source_code,
-        maybe_output_code: None,
-        maybe_source_map: None,
-      }));
-    }
-    Ok(None) // cannot find locally
+    let p = Path::new(&filename);
+    let media_type_filename = [&filename, ".mime"].concat();
```

Comment:
This has a very interesting implication: if a user provides a `.mime` file containing typescript mime type, s/he can use the extensionless file as if it has the .ts extension...

Reply:
Hmm do we even need the mime type files anymore ?

Reply:
But the compiler will have issues with it (e.g. why the shebang scripts don't work without extensions at the moment).

Reply:
I'd like to clean up the mime file usage in the future. It seems we don't need most of them. Leaving as is for now.